spec.md

```diff
@@ -494,6 +503,39 @@ message PluginCapability {
 
 If the plugin is unable to complete the GetPluginCapabilities call successfully, it MUST return a non-ok gRPC code in the gRPC status.
 
+#### `Probe`
+
+A Plugin MUST implement this RPC call.
+The primary utility of the Probe RPC is deployment verification: this can be a one time or periodic check to ensure that a plugin instance is healthy and has all dependencies available.
+This information can be used, for example, to monitor the health of the plugin and redeploy the plugin (or take other automated measures) when it becomes unhealthy.
+
+The Plugin SHOULD verify if it has the right configurations, devices, dependencies and drivers in order to run and return a success if the validation succeeds.
+The CO SHALL invoke this RPC prior to any other controller service or node service RPC in order to allow the CO to determine the readiness of the plugin.
+A CO MAY invoke this call multiple times with the understanding that a plugin's implementation MAY NOT be trivial and there MAY be overhead incurred by such repeated calls.
+The SP SHALL document known limitations regarding a particular Plugin's implementation of this RPC.
+For example, the SP MAY document the maximum frequency at which its Probe implementation should be called.
+
+```protobuf
+message ProbeRequest {
+  // The API version assumed by the CO. This is a REQUIRED field.
+  Version version = 1;
+}
+
+message ProbeResponse {}
+```
+
+##### Probe Errors
```

Comment:
Wondering if we should have use a RESOURCE_EXHAUSTED code if the plugin thinks that Probe() is being called too frequently?